src/nvim/strings.c

```diff
@@ -394,72 +394,28 @@ int vim_strnicmp(const char *s1, const char *s2, size_t len)
 }
 #endif
 
-/*
- * Version of strchr() and strrchr() that handle unsigned char strings
- * with characters from 128 to 255 correctly.  It also doesn't return a
- * pointer to the NUL at the end of the string.
- */
-char_u *vim_strchr(const char_u *string, int c)
-  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+/// strchr() version which handles multibyte strings
+///
+/// @param[in]  string  String to search in.
+/// @param[in]  c  Character to search for. Must be a valid character.
+///
+/// @return Pointer to the first byte of the found character in string or NULL
+///         if it was not found. NUL character is never found, use `strlen()`
+///         instead.
+char_u *vim_strchr(const char_u *const string, const int c)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  int b;
-
-  const char_u *p = string;
-  if (enc_utf8 && c >= 0x80) {
-    while (*p != NUL) {
-      int l = (*mb_ptr2len)(p);
-
-      // Avoid matching an illegal byte here.
-      if (l > 1 && utf_ptr2char(p) == c) {
-        return (char_u *) p;
-      }
-      p += l;
-    }
+  assert(c >= 0);
```

Comment:
Given the next line it looks like this is a leftover of when I thought that c is never NUL.